production-vms/config/azure.go

```diff
@@ -0,0 +1,154 @@
+package config
+
+import (
+	"crypto/tls"
+	"fmt"
+	"net"
+	"net/http"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/arm"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/compute/armcompute"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/network/armnetwork"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/resources/armresources"
+	"github.com/Azure/go-armbalancer"
+)
+
+// modeled after e2e/config/azure.go
```

Comment:
I think we could think about exporting this to a common package, I was thinking the same for all the code around parsing/reading components.json as well

Reply:
Yeah I think there is a bigger question around - do we want it to be more readable on a project-based-level or is readability kind of a non-issue for this area of code and its better to be more common/dry? I just finished that go course you recommended forever ago (slowly but surely lol) and it had me thinking about the package management / file structure lol

Reply:
what is this go course i should take it too

Reply:
its the ultimate-go-programming course second edition - https://learning.oreilly.com/course/ultimate-go-programming/9780135261651/  you should be able to get it for free using your MS login on oreilly's 

warning that it is LONG it took me months to get through it, kind of getting motivation to watch bursts of it since every moment is filled with information. So it's a really good watch, but take it at your own pace because it really is an overload of information!

Reply:
@element-of-surprise also has an awesome-looking book on go for devops that I've been meaning to read as well :)

Reply:
definitely would like to see code reuse!